Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -303,6 +313,11 @@ void MediaPlayerPrivateGStreamerMSE::updateStates()
             GST_ERROR_OBJECT(pipeline(), "Setting the pipeline to PAUSED failed");
         m_isPipelinePlaying = false;
     }
+
+    if (!m_isPlaybackRatePaused && getStateResult == GST_STATE_CHANGE_SUCCESS && state >= GST_STATE_PAUSED) {
+        // Update playback rate that was set before the pipeline was prerolled.
+        updatePlaybackRate();
```

Comment:
What does `updatePlaybackRate()` and in what situations do we need to call it?

Reply:
`updatePlaybackRate()` accounts for `m_playbackRatePaused` and moves the pipeline to playing if needed, among other things. That's why we need to call it in this situation.

Reply:
But isn't that already the responsibility of `updateStates()`, just in the previous if() statement?